From a2272df1a0bb7ebf5707dde15de7e6a7be7724d4 Mon Sep 17 00:00:00 2001
From: Ilya Leoshkevich <45998171+iii-i@users.noreply.github.com>
Date: Wed, 4 Mar 2020 18:35:33 +0100
Subject: [PATCH] common: add big-endian support to Ip6ntohl and Ip6htonl
 (#10250)

This patch resolves the TODO from
https://github.com/envoyproxy/envoy/pull/2431.
Tests pass on a big-endian machine (IBM Z).

Signed-off-by: Ilya Leoshkevich <iii@linux.ibm.com>
---
 source/common/network/utility.cc | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/source/common/network/utility.cc b/source/common/network/utility.cc
index f41e306f19f..fd920c67ef4 100644
--- a/source/common/network/utility.cc
+++ b/source/common/network/utility.cc
@@ -424,17 +424,19 @@ bool Utility::portInRangeList(const Address::Instance& address, const std::list<
 }
 
 absl::uint128 Utility::Ip6ntohl(const absl::uint128& address) {
-  // TODO(ccaraman): Support Ip6ntohl for big-endian.
-  static_assert(ABSL_IS_LITTLE_ENDIAN,
-                "Machines using big-endian byte order is not supported for IPv6.");
+#ifdef ABSL_IS_LITTLE_ENDIAN
   return flipOrder(address);
+#else
+  return address;
+#endif
 }
 
 absl::uint128 Utility::Ip6htonl(const absl::uint128& address) {
-  // TODO(ccaraman): Support Ip6ntohl for big-endian.
-  static_assert(ABSL_IS_LITTLE_ENDIAN,
-                "Machines using big-endian byte order is not supported for IPv6.");
+#ifdef ABSL_IS_LITTLE_ENDIAN
   return flipOrder(address);
+#else
+  return address;
+#endif
 }
 
 absl::uint128 Utility::flipOrder(const absl::uint128& input) {
